### PR TITLE
[feature] logo item animator

### LINKED ISSUE #347: Inventory: Item logo on HUD
Real dan Astral gak beda aset

Use -> aktif di environment (Evidence) -> 1 logo gak berubah (sama kayak logo gak aktif) + ada visual effects/audio di itemnya pas diaktifin dia ngapain
Use -> di consume -> deactivate/removed (Sanskrit) -> 2 logo (udah di consume/belom)
Use -> aktif di tangan (Ankh) -> 2 logo (aktif/gak aktif)

Total 4 logo

Add animator instead of 2 images for the HUD

### LINKED ISSUE #355: [Refactor] Inventory and HUD Observer flow
use string key + generic parameter

Jadiin satu event:
- ItemLogoEvent
- InitItemHudEvent
- ToggleItemHudDisplayEvent
- UpdateActiveItemIndexEvent

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- Implement `animator` and `animations` for ankh and evidence item hud logo
- Integrate item logo animator to item script and pass it to hudmanager through inventory script
- Add new field `LogoState` in item script as reference to animator parameter

### Tasks List
- [close #347 ]
- [close #355 ]

### Attachments
- Ankh Logo Example

https://user-images.githubusercontent.com/51603162/176921885-5b51b212-b0ba-4820-9c4c-276d91db8c25.mp4

- Evidence Logo Example

https://user-images.githubusercontent.com/51603162/176921955-f7653eef-dab5-4b1e-9bc0-790ffab08c3a.mp4

### CODE REVIEW

File: Assets/Astralization/Scripts/Item/EvidenceItem/ClockItem.cs [hunk 1]
```diff
@@ -26,12 +26,14 @@ protected override void Awake()
...
     {
         _clockManager.ChangeState<ClockActiveState>();
+        LogoState = _clockManager.GetStateNum();
     }
 
     public override void Pick()
     {
         base.Pick();
-        _clockManager.ChangeState<ClockInactiveState>();
+        //_clockManager.ChangeState<ClockInactiveState>();
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Item/EvidenceItem/ClockItem.cs:35]: Why is this commented? If its not needed I think deleting it is fine

File: Assets/Astralization/Scripts/Item/EvidenceItem/SilhouetteBowlItem.cs [hunk 2]
```diff
@@ -23,12 +23,14 @@ protected override void Awake()
...
     {
         _silhouetteBowlManager.ChangeState<SilhouetteBowlActiveState>();
+        LogoState = _silhouetteBowlManager.GetStateNum();
     }
 
     public override void Pick()
     {
         base.Pick();
-        _silhouetteBowlManager.ChangeState<SilhouetteBowlInactiveState>();
+        //_silhouetteBowlManager.ChangeState<SilhouetteBowlInactiveState>();
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Item/EvidenceItem/SilhouetteBowlItem.cs:32]: Same for this line, delete if unnecessary

File: Assets/Astralization/Scripts/Item/EvidenceItem/ThermometerItem.cs [hunk 3]
```diff
@@ -22,12 +22,14 @@ protected override void Awake()
...
     {
         _thermometerManager.ChangeState<ThermometerActiveState>();
+        LogoState = _thermometerManager.GetStateNum();
     }
 
     public override void Pick()
     {
         base.Pick();
-        _thermometerManager.ChangeState<ThermometerInactiveState>();
+        //_thermometerManager.ChangeState<ThermometerInactiveState>();
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Item/EvidenceItem/ThermometerItem.cs:31]: Same as this line

File: Assets/Astralization/Scripts/Player/Inventory.cs [hunk 4]
```diff
@@ -1,10 +1,61 @@
 using System;
 using UnityEngine;
 
+#region EventArgs
+// TODO: pindahin ke utils
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Player/Inventory.cs:5]: Do we need to add this to backlog?

**Review by InShineStudios — CHANGES_REQUESTED** [review]

File: Assets/Astralization/Scripts/Item/EvidenceItem/ClockItem.cs (hunk 1 above)
Comment (rafiftaris) [reply Assets/Astralization/Scripts/Item/EvidenceItem/ClockItem.cs:35]: Setting EvidenceItem to its negative state is **temporarily** disabled so the item logo can show active/positive/negative states when picked up. So is it better to just delete it?

File: Assets/Astralization/Scripts/Player/Inventory.cs (hunk 4 above)
Comment (rafiftaris) [reply Assets/Astralization/Scripts/Player/Inventory.cs:5]: Sorry, I should have moved it since the ItemHelper.cs is already up to dev. Will fix later

**Review by RaiRiz — APPROVED** [review]
LGTM, fungsionalitas sudah jalan sesuai ekspektasi dan test juga semua success

**Review by naufalhilmi12345 — APPROVED** [review]
LGTM! cuman mau nanya buat evience item, sekali dipake itu gabisa di non-aktifin ya?? jadi akan selalu item setelah dipake pertama kali?? thanks

File: Assets/Astralization/Scripts/Item/EvidenceItem/ClockItem.cs (hunk 1 above)
Comment (rafiftaris) [reply Assets/Astralization/Scripts/Item/EvidenceItem/ClockItem.cs:35]: tandain biar diuncomment

**Pull request merged**